cluster/search/load_data.py

```diff
@@ -32,71 +42,171 @@ def parse_obj_to_document(obj):
+    #         new_doc["indico"]["title_places"] = text_analysis.get('places')[1]
+    #         new_doc["indico"]["title_organizations"] = text_analysis.get('organizations')[1]
+    #         new_doc["indico"]["title_text_features"] = text_analysis.get('text_features')[1]
+
+
+    #         new_doc["indico"]["sentiment"] = text_analysis.get('sentiment_hq')[0]
+    #         new_doc["indico"]["keywords"] = text_analysis.get('keywords')[0]
+    #         new_doc["indico"]["people"] = text_analysis.get('people')[0]
+    #         new_doc["indico"]["places"] = text_analysis.get('places')[0]
+    #         new_doc["indico"]["organizations"] = text_analysis.get('organizations')[0]
+    #         new_doc["indico"]["text_features"] = text_analysis.get('text_features')[0]
+
+
+    #         new_documents.append(new_doc)
+    #         new_doc = {}
+    #     except:
+    #         import traceback; traceback.print_exc()
+    #         return False
+
     return new_documents
 
-def upload_data(data_dir, filename):
+
+def upload_data(es, current_dir):
     t0 = time.time()
+    executor = ThreadPoolExecutor(max_workers=4)
+    all_files = []
+    for subdir, dirs, files in os.walk(current_dir):
+        for filename in files:
+            all_files.append(os.path.join(current_dir, filename))
+
+    def sub_problem(data_file):
+        print data_file
+        lines = [line for line in pe.get_records(file_name=data_file) if line.get("description_text") and len(line.get("description_text")) > 600][:100]
+        print len(lines)
+
+
+        documents = map(lambda x: parse_obj_to_document(x), lines)
+        documents = filter(lambda doc: doc.link and "Service Unavailable" not in doc.text, documents)
+        print 'raw docs'
+        return add_indico(documents)
 
-    documents = []
-    data_file = os.path.join(data_dir, filename)
-    with open(data_file, 'rb') as f:
-        documents = [json.loads(l) for l in f.readlines()]
-
-    # print len(lines)
-    # documents = map(lambda x: parse_obj_to_document(json.loads(x)), lines)
-    # documents = filter(lambda doc: doc.link and "Service Unavailable" not in doc.text, documents)
-    
-    clean_documents = add_indico(documents, filename)
-    
-    data_dir = os.path.join(os.path.dirname(__file__), '../../entities/')
-    if clean_documents:
-        print "about to write", len(clean_documents), "documents to", data_dir
+    futures = {executor.submit(sub_problem, data_file): data_file for data_file in all_files}
+    data_dir = os.path.join(os.path.dirname(__file__), '../../backups/')
+    for future in concurrent.futures.as_completed(futures):
+        clean_documents = futures[future]
         with open(data_dir+filename, 'w') as data_dump:
             for document in clean_documents:
                 data_dump.write(json.dumps(document)+'\n')
 
-    print 'done - total time:', time.time() - t0
-    return True
-            
+        print "written"
+        # es.upload(clean_documents)
```

Comment:
Needs to be uncommented eventually, but I don't want to fuck with the database while I'm playing with this unfinished code.